core/src/dbs/store.rs

```diff
@@ -0,0 +1,53 @@
+use crate::dbs::plan::Explanation;
+use crate::sql::value::Value;
+use std::cmp::Ordering;
+use std::mem;
+
+#[derive(Default)]
+// TODO Use surreal-kv once the number of record reach a given threshold
+pub(super) struct StoreCollector(Vec<Value>);
+
+impl StoreCollector {
+	pub(super) fn push(&mut self, val: Value) {
+		self.0.push(val);
+	}
+
+	// When surreal-kv will be used, the key will be used to sort the records in surreal-kv
```

Comment:
Excellent work, as always!

It is not important, but probably for the future; is there any reason you mention surreal-kv here?

To me, sorting a massive set of values sounded out of the scope of a KVS, especially when it comes to distributed sorting in a multi-node situation.

Reply:
When using SELECT/ORDER, actually the records are stored in memory and sorted in memory. Let's say you are using FDB or TIKV and you have billions of records, there is a lot of chance the SurrealDB node will consume all of the memory and end up with an OOM.

The idea is indeed not to push bash the records to a remote KV store but to store the records locally in the coordinator node so they can be sorted. Like a temporary table, the SurealDB coordinator node can store the records locally in a local KV (it could have been RocksDB). Then the sorting strategy could be based on the key, or by using a secondary index (based on the existing BTree implementation) .

Reply:
> store the records locally in the coordinator node so they can be sorted

@emmanuel-keller Neat!

I blindly assumed we would do that with the main kvs(TiKV, FDB, and so on in a distributed setup). But offloading to a local disk is preferable if the temporary table fits within the local disk, even if it's a multi-node deployment! Makes sense.

Thank you very much for clarifying 🙏